sdk/servicebus/service-bus/src/serviceBusClient.ts

```diff
@@ -149,33 +204,6 @@ export class ServiceBusClient {
     }
   }
 
-  /**
-   * Creates a ServiceBusClient for the Service Bus Namespace represented in the given connection
-   * string.
-   * @param {string} connectionString - Connection string of the form
-   * 'Endpoint=sb://my-servicebus-namespace.servicebus.windows.net/;SharedAccessKeyName=my-SA-name;SharedAccessKey=my-SA-key'
-   * @param {ServiceBusClientOptions} [options] Options to control ways to interact with the
-   * Service Bus Namespace.
-   * @returns {ServiceBusClient}
-   */
-  static createFromConnectionString(
-    connectionString: string,
-    options?: ServiceBusClientOptions
-  ): ServiceBusClient {
-    const config = ConnectionConfig.create(connectionString);
-
-    config.webSocket = options && options.webSocket;
-    config.webSocketEndpointPath = "$servicebus/websocket";
-    config.webSocketConstructorOptions = options && options.webSocketConstructorOptions;
-
-    // Since connectionstring was passed, create a SharedKeyCredential
-    const credential = new SharedKeyCredential(config.sharedAccessKeyName, config.sharedAccessKey);
-
-    ConnectionConfig.validate(config);
-
-    return new ServiceBusClient(config, credential, options);
-  }
-
   //   /**
   //    * Creates a ServiceBusClient for the Service Bus Namespace represented by the given host using
   //    * the given TokenProvider.
```

Comment:
The constructor overloads that support `TokenCredential`  replace the `createFromTokenProvider` and `createFromAadTokenCredentials` methods that are commented here. These commented lines can now be removed.

The [tests for these 2 methods](https://github.com/Azure/azure-sdk-for-js/blob/9883ac911b4c9cfadd9e8d06c4837eb58c007106/sdk/servicebus/service-bus/test/serviceBusClient.spec.ts#L312-L384) that were commented need to be replaced with tests that use the `EnvironmentCredential`. See [similar tests for Event Hubs](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/test/client.spec.ts#L88-L127) for reference

Reply:
Using `@azure/identity` in tests is causing build and rollup to fail.
Specifically, seems to do with resolving references to `RequestInit` `RequestInfo` `Response` `Headers` in `core-http` which seem to have been auto inferred from `NodeJS.Globals` / `@types/node-fetch`

Just added a draft commit to this PR if anyone is available to take a look, as this will likely involve more investigation, troubleshooting
cc: @bterlson 

The errors I'm seeing are

```
tsc -p . && rollup -c 2>&1 && npm run extract-api
../../core/core-http/es/lib/fetchHttpClient.d.ts(7,72): error TS2304: Cannot find name 'RequestInit'.
../../core/core-http/es/lib/fetchHttpClient.d.ts(9,27): error TS2304: Cannot find name 'RequestInfo'.
../../core/core-http/es/lib/fetchHttpClient.d.ts(9,47): error TS2304: Cannot find name 'RequestInit'.
../../core/core-http/es/lib/fetchHttpClient.d.ts(9,69): error TS2304: Cannot find name 'Response'.
../../core/core-http/es/lib/fetchHttpClient.d.ts(11,47): error TS2304: Cannot find name 'Headers'.
../../core/core-http/es/lib/nodeFetchHttpClient.d.ts(7,18): error TS2304: Cannot find name 'RequestInfo'.
../../core/core-http/es/lib/nodeFetchHttpClient.d.ts(7,38): error TS2304: Cannot find name 'RequestInit'.
../../core/core-http/es/lib/nodeFetchHttpClient.d.ts(7,60): error TS2304: Cannot find name 'Response'.
../../core/core-http/es/lib/nodeFetchHttpClient.d.ts(8,63): error TS2304: Cannot find name 'RequestInit'.
```

Reply:
After updating the version of the `@azure/identity` to `^1.0.0` in the package.json file, I am able to build without any errors:

![image](https://user-images.githubusercontent.com/16890566/72115910-79373f80-32fd-11ea-8080-844889a9eaef.png)

Reply:
Updated PR and verified test run on local

Reply:
The commented code for the static constructors in this file still needs to be removed

Reply:
updated